### PR TITLE
[cudagraphs] static input params flag

### PR DESCRIPTION
**Pull request by xmfan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127006
* #127960
* __->__ #127668
* #125382

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by BoyuanFeng — COMMENTED** [review]
A few minor comments. Please also add proper tests.

File: torch/_inductor/cudagraph_trees.py [hunk 1]
```diff
@@ -2045,6 +2045,12 @@ def add_function(
         placeholders,
         mutated_input_idxs,
     ) -> Tuple[Callable[..., Any], List[Optional[Tensor]]]:
+        if torch._inductor.config.triton.cudagraph_static_input_params:
+            all_static_input_idxs = set(static_input_idxs)
+            for i, inp in enumerate(inputs):
+                if isinstance(inp, torch.nn.Parameter):
```
[the remark this answers is not in the document]
Comment (xmfan) [reply torch/_inductor/cudagraph_trees.py:2051]: another case is hierarchical compile. we will need to move away from the flag

**Comment by eellison** [comment]
re-request when ready

File: torch/_inductor/cudagraph_trees.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xmfan) [reply torch/_inductor/cudagraph_trees.py:2051]: actually hierarchichal compile is doing the exact same thing 😆 
https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/collect_metadata_analysis.py#L669C33-L674

**Pull request closed without merging**